### PR TITLE
Clustering SPI update

### LINKED ISSUE #104: testFaultTolerance failures
reported at : https://travis-ci.org/vert-x3/vertx-hazelcast/jobs/452894547

```
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 109.352 sec <<< FAILURE! - in io.vertx.core.eventbus.HazelcastFaultToleranceTest
testFaultTolerance(io.vertx.core.eventbus.HazelcastFaultToleranceTest)  Time elapsed: 109.349 sec  <<< FAILURE!
java.lang.AssertionError: Survivor pongs expected:<10> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at io.vertx.test.core.AsyncTestBase.assertEquals(AsyncTestBase.java:363)
	at io.vertx.core.eventbus.FaultToleranceTest.assertEqualsEventually(FaultToleranceTest.java:144)
	at io.vertx.core.eventbus.FaultToleranceTest.testFaultTolerance(FaultToleranceTest.java:89)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:108)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:78)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:54)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:144)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:203)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
```

### PR DESCRIPTION
**Pull request by tsegismont**
Follows-up on https://github.com/eclipse-vertx/vert.x/pull/3132

Closes #104

### CODE REVIEW

**Comment by tsegismont** [comment]
Sharing a draft PR for now. `FaultToleranceTest` is failing so far.

**Comment by tsegismont** [comment]
Test suite passing on my laptop now. Still some work to do so not yet ready for thorough review.

File: src/test/resources/cluster.xml [hunk 1]
```diff
@@ -90,15 +90,7 @@
...
-  <multimap name="__vertx.subs">
-
-    <!--
-        Number of backups. If 1 is set as the backup-count for example,
-        then all entries of the map will be copied to another JVM for
-        fail-safety. 0 means no backup.
-    -->
-    <backup-count>1</backup-count>
-  </multimap>
+  <replicatedmap name="__vertx.subs"/>
```
Comment (vietj) [inline src/test/resources/cluster.xml:93]: why is this removed ?

Comment (tsegismont) [reply src/test/resources/cluster.xml:93]: The subs will now be stored in a replicated map instead of a multimap. At the expense of a bigger memory footprint, it makes storage reliable and relieves the implementation from publishing subs again when topology changes.

**Comment by tsegismont** [comment]
@vietj PTAL
